### PR TITLE
Initial packaging

### PR DESCRIPTION
**Pull request by leonheldattoradex**
From meta-imx@kirkstone-5.15.71-2.2.2

This can't really fit in our packing infrastructure because there's no source code, just a binary script. This is ok, we can have these "special" packages like this, as they're mostly slow-changing.

Related-to: TCCP-21

### CODE REVIEW

File: debian/control [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Build-Depends: debhelper (>= 9),
+               lsb-release,
+               wget
+Standards-Version: 4.5.1
+Homepage: https://www.nxp.com/lgfiles/NMG/MAD/YOCTO/libgpuperfcnt-6.4.3.p4.2-aarch64-wayland.bin
+
+Package: libgpuperfcnt
+Architecture: arm64
+Depends: ${shlibs:Depends}, ${misc:Depends}, imx-gpu-viv-wayland
+Description: libgpuperfnct expose the status GPUs found in iMX SoCs
```
Comment (embdur) [inline debian/control]: https://github.com/Freescale/meta-freescale/blob/0870f92ab9cf685d22f40f5e377f7de2e1c591dd/SCR/SCR-5.15.71-2.2.0.txt#L756

> Library to retrieve i.MX GPU performance data

Comment (embdur) [reply debian/control]: Just bikeshedding over here, LGTM

**Review by embdur — APPROVED** [review]
Looks good, thanks!

**Comment by charles2910** [comment]
Hi, sorry for being a little late for the party.

I tried to build the package via the regular `sbuild` and got no luck:


<details><summary>sbuild log</summary>

```bash
dpkg-buildpackage
-----------------

Command: dpkg-buildpackage --sanitize-env -us -uc -rfakeroot
dpkg-buildpackage: info: source package libgpuperfcnt
dpkg-buildpackage: info: source version 6.4.3
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Leonardo Held <leonardo.held@toradex.com>
 dpkg-source --before-build .
dpkg-buildpackage: info: host architecture amd64
 fakeroot debian/rules clean
dh clean
 dpkg-source -b .
dpkg-source: warning: no source format specified in debian/source/format, see dpkg-source(1)
dpkg-source: info: using source format '1.0'
dpkg-source: info: building libgpuperfcnt in libgpuperfcnt_6.4.3.tar.gz
dpkg-source: info: building libgpuperfcnt in libgpuperfcnt_6.4.3.dsc
 debian/rules build
dh build
 fakeroot debian/rules binary
dh binary
 dpkg-genbuildinfo -O../libgpuperfcnt_6.4.3_amd64.buildinfo
dpkg-genbuildinfo: error: binary build with no binary artifacts found; .buildinfo is meaningless
dpkg-buildpackage: error: dpkg-genbuildinfo -O../libgpuperfcnt_6.4.3_amd64.buildinfo subprocess returned exit status 25
--------------------------------------------------------------------------------
Build finished at 2024-01-16T13:38:18Z
```
</details>

1. I saw the workflow to build the package is working, so I wonder what was the build command that made it work for you - it would be nice to document it in a `README.Debian` file.

2. You can get rid of the `debian/compat` file declaring a build dependency on `debhelper-compat (=13)` in the `debian/control` file.

3. A long description in `debian/control` is a nice to have, though completely superfluous in this case. Something in the lines of the pkgconfig would work "Abstraction library used to retrieve HW counter values, and registers from Vivante/VSI GPU drivers".

4. Declaring source package level 3 compatibility is important (creating `debian/source/format` with `3.0 (quilt)`) is important because it defaults to version 1.0 (which is very much legacy by now) when nothing is declared.

5. Usually on Debian we break down libraries in two packages, a -dev with the header files and `.so` symlink and another ending with the major so version. In this case would be `libgpuperfcnt1` and `libgpuperfcnt-dev` packages.

6. Are we having the symbols files for this one?

7. the pkgconfig provided by NXP seems buggy, should/do we care?

**Pull request merged**